### ISSUE TITLE
Combinable filter capabilities

### ISSUE THREAD

**Issue by eltos** [pulsejet/memories#868]
**Is your feature request related to a problem? Please describe.**

There is currently no generic filter capability implemented. A handful of specific filters are available from the side menu (favourites, person, place, tag), **but these filters cannot be combined**. E.g. it is not possible to
- Show images with both, Alice and Bob
- Show images from Aug 2023 on the map
- Show only videos in a specific album
- Show images with cats in LA
- ...

**Describe the solution you'd like**
A generic method to filter photos in any of the views (most importantly the timeline and maps views, but also folders, videos, albums etc.). And to combine these filters.
This should allow filtering by
- Date and time (year/month or from ... to)
- Location(s) (with hierarchical country/area/town options, like currently available in the places view, also an option for "missing GPS")
- Content type (photo, video, gif, live photo, panorama, 360°, ...), see #676
- Person(s) involved, see #673
- Favourite/Rating/Stars, see #564
- Tag(s)/keyword(s)
- (Optionally) other EXIF fields, e.g. resolution, camera model, orientation
- (Optionally) shared by/with user

Probably a bar at the top providing dropdowns/sliders/etc. would be a good way to do this.

This would also provide some useful tools to maintain one's image library, e.g. to filter images which are missing GPS tags, 

**Describe alternatives you've considered**

- Specific solutions for each of the above-mentioned aspects (see linked issues) as separate categories in the menu. But this would not make for a generic filter capability and not allow for combinations of filters.
- Using Nextcloud search functionality instead of filtering. This would be less intuitive and much more complicated to implement

**Additional context**

Requires same/similar backend implementations as for #352

Slider to filter by date and time used by the [Maps app](https://github.com/nextcloud/maps)
![grafik](https://github.com/pulsejet/memories/assets/19860638/e211bb6f-cb60-4b7d-b26c-c6f944d71b7f)

Filter by country, area, place, etc. used by the places view
![grafik](https://github.com/pulsejet/memories/assets/19860638/9d66a794-b9bc-487e-a2ff-3be1d306afdc)

**Comment by jovanbulck** [comment]
I wanted to share a quick update regarding the filtering feature, as I found it to be quite essential. I have implemented basic support for filtering, which also includes some functionalities outlined in issues #1004, #1238, and #1042.

This implementation builds upon the user labeling code from PR #1266. Once that PR is merged, I would be happy to open a new PR specifically for the filtering feature.

Here’s a preview of the current implementation, feedback is always welcome :) :

![image](https://github.com/user-attachments/assets/aab5b39c-7bcf-460d-9dec-ebfc4faf16a8)

**Comment by amaciuc** [comment]
Will this also be available on mobile app ?

**Comment by jovanbulck** [comment]
> Will this also be available on mobile app ?

yes, this is currently also implemented with mobile support on my fork, if you're interested you can find the code here:

https://github.com/jovanbulck/memories/commit/ad64ac594388895a79efa18ad8a21754dc322740

I'll need to find some time to rebase and update this now that #1266 is merged to open a PR to get this upstream :)

**Comment by pktiuk** [comment]
This is visualization of my suggestion:

![Image](https://github.com/user-attachments/assets/fcc7a109-b681-4841-b67d-90657b0dcbbd)
